### PR TITLE
update rules table style + modal setup

### PR DESCRIPTION
**Pull request by rohanmathur91**
Closes issue: 

## 📜 Summary of changes:
- Updated base styles for rules table
- Added rules context
- Setup more record actions
- Setup Modals for single row action

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx [hunk 1]
```diff
@@ -30,31 +60,133 @@ const useRuleTableColumns = () => {
...
+              type="text"
+              icon={<MdOutlineShare />}
+              onClick={() => {
+                handleRuleShare(record);
+              }}
+            />
+
+            <Dropdown menu={{ items: recordActions }} trigger={["click"]} overlayClassName="rule-more-actions-dropdown">
+              <Button type="text" className="more-rule-actions-btn" icon={<MdOutlineMoreHoriz />} />
+            </Dropdown>
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx:166]: We should create a separate component like `RuleTableActions` which has this rendering logic and then assign this component to `render` property of this col. We may have more actions in future

File: app/src/componentsV2/ContentTable/contentTable.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+  }
+}
+
+tr.rq-content-table-row.ant-table-row {
+  background: var(--background);
+
+  &.ant-table-row-selected {
+    background: var(--Highlight-dark-bg, #111a2c);
+    border-bottom: 1px solid var(--surface-1, #282828);
```
Comment (RuntimeTerror10) [inline app/src/componentsV2/ContentTable/contentTable.scss:42]: We can remove the fallback color here

File: app/src/componentsV2/ContentTable/contentTable.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      }
+    }
+  }
+}
+
+tr.rq-content-table-row.ant-table-row {
+  background: var(--background);
+
+  &.ant-table-row-selected {
+    background: var(--Highlight-dark-bg, #111a2c);
```
Comment (RuntimeTerror10) [inline app/src/componentsV2/ContentTable/contentTable.scss:41]: Shouldn't  we add this variable if it's not defined yet?

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx [hunk 4]
```diff
@@ -30,31 +60,133 @@ const useRuleTableColumns = () => {
...
+      key: "modificationDate",
+      align: "center",
+      width: 152,
+      responsive: ["lg"],
+      render: (rule: RuleTableDataType) => {
+        if (rule.objectType && rule.objectType === RuleObjType.GROUP) {
+          return null;
+        }
+        const dateToDisplay = rule.modificationDate ? rule.modificationDate : rule.creationDate;
+        const beautifiedDate = moment(dateToDisplay).format("MMM DD");
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx:99]: Add year too

**Comment by RuntimeTerror10** [comment]
contents of `updated on` column are center aligned , and other columns are left aligned

Screenshot 2023-11-09 at 1 19 13 PM

**Review by RuntimeTerror10 — CHANGES_REQUESTED** [review]

File: app/src/componentsV2/ContentTable/ContentTable.tsx [hunk 5]
```diff
@@ -30,16 +29,29 @@ const ContentTable = <DataType extends object>({
     setFilteredRowsData([...data]);
   }, [data]);
 
+  useEffect(() => {
+    bulkActionBarConfig?.options?.getSelectedRowsData?.(selectedRowsData);
```
Comment (wrongsahil) [inline app/src/componentsV2/ContentTable/ContentTable.tsx:33]: .

**Review by wrongsahil — CHANGES_REQUESTED** [review]

**Pull request merged**